### PR TITLE
DAOS-6375 control: Enforce system name checking in RPCs

### PR DESCRIPTION
**Pull request by mjmac**
Control plane RPCs added over time have been inconsistent about
including the system name, leading to a situation where many
RPCs may be invoked for the wrong system. Adds missing fields
to protobuf messages and a centralized method for checking that
the request's system matches the running system.

### CODE REVIEW

**Comment by mjmac** [comment]
@liw: Tagged you for comment on the overall goal here. I realized that we haven't been consistent about sending the system name as part of the RPC payloads, nor have we been consistently checking the name, so that should be fixed by this PR. Just want to verify that this is the right approach.

**Comment by liw** [comment]
> @liw: Tagged you for comment on the overall goal here. I realized that we haven't been consistent about sending the system name as part of the RPC payloads, nor have we been consistently checking the name, so that should be fixed by this PR. Just want to verify that this is the right approach.

My thought had always been yes we should do so, until you inspired some new thoughts. :)

Between apps and daos_agent, a system identifier _is a must_. We have the following options:

- Unix domain socket path. This allows an alternative, one-agent-per-system architecture. Such paths, however, may be long and confused with the paths in the DAOS global name space.
- System name. This dictates a one-agent-multi-system architecture. daos_agent needs to interoperate with multiple systems, though I think given the mandatory single-system upgrade/downgrade interoperability requirement this may not an extra requirement. Such names are more user friendly (especially to Lustre users). (The Unix domain socket path environment variable in this case should be transparent most of the time, like GOROOT or GOPATH.)

In contrary, between daos_agent/dmg and daos_server, a system identifier _is not a must_, as the host:port and the certificate should provide two checks already (correct?). I'm not against adding a third check for consistency, as long as it will not be a burden on users (e.g., on command lines).

Just for considerations.

**Comment by mjmac** [comment]
> I'm not against adding a third check for consistency, as long as it will not be a burden on users (e.g., on command lines).
> 
The server/client code uses a default system name is one is not provided via configuration, so I don't think that this would add any extra burden. I think my primary question at this point is: Can anyone think of a reason why the approach in this PR might be a bad idea? I want to get most of our protobuf churn out of the way before 1.2 is released because we're still in a grace period where we don't need to worry about interoperability.

**Review by tanabarr — APPROVED** [review]

File: src/control/cmd/dmg/pool.go [hunk 1]
```diff
@@ -132,9 +132,10 @@ func (cmd *PoolCreateCmd) Execute(args []string) error {
 
 	req := &control.PoolCreateReq{
 		ScmBytes: scmBytes, NvmeBytes: nvmeBytes, Ranks: ranks,
-		NumSvcReps: cmd.NumSvcReps, Sys: cmd.Sys,
-		User: cmd.UserName, UserGroup: cmd.GroupName, ACL: acl,
+		NumSvcReps: cmd.NumSvcReps,
+		User:       cmd.UserName, UserGroup: cmd.GroupName, ACL: acl,
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/cmd/dmg/pool.go:136]: You'll have to take that up with gofmt. ;)

**Review by kjacque — APPROVED** [review]

**Pull request merged**